src/corehost/cli/fxr/hostfxr.cpp

```diff
@@ -566,3 +395,54 @@ SHARED_API hostfxr_error_writer_fn hostfxr_set_error_writer(hostfxr_error_writer
 {
     return trace::set_error_writer(error_writer);
 }
+
+//
+// Gets a typed delegate to perform an action on the currently loaded CoreCLR or on a newly created one.
+//
+// Parameters:
+//     libhost_path
+//          Absolute path of the entry hosting library
+//     dotnet_root
+//     type
+//          The requested delegate type
+//     delegate
+//          Requested delegate.
+// Return value:
+//     The error code result.
+//
+// A new CoreCLR instance will be created or reused if the existing instance can satisfy the configuration
+// requirements supplied by the runtimeconfig.json file.
+//
+SHARED_API int32_t hostfxr_get_coreclr_delegate(
+    const pal::char_t* libhost_path,
```

Comment:
This is an export, correct?  When adding a new export we should always keep in mind 'what future proofing should we do?'  This may be the ultimate set of parameters, but if it is not we are likely stuck with a hostfxr_get_coreclr_delegate2 in the future.

Reply:
Yep. This is something I am looking into for a slight modification for IJW work. There is a small chance one more argument is added and I am working on that now.

Reply:
For IJW, I'd think we'd want something that starts the runtime and maybe loads an assembly, but not a delegate back.

Reply:
@morganbr There is no assumption about the specifics of the delegate in this case. I realize whenever someone hears 'delegate' they think managed delegate. But consider the signature of the function, it is returning a function pointer. It could be a function pointer to an internal native function or a managed function, literally anything. Also, the enum should be observed which will indicate what scenario the delegate is supposed to be supporting. For IJW, this could be a function pointer that takes 100 arguments or none. It is up to us and what we need.